nautobot/core/ui/object_detail.py

```diff
@@ -713,7 +821,10 @@ def __init__(
                 - `[render_markdown, placeholder]` - render the given text as Markdown, or render a placeholder if blank
                 - `[humanize_speed, placeholder]` - convert the given kbps value to Mbps or Gbps for display
         """
+        if data and context_data_key:
+            raise ValueError("The data and context_data_key parameters are mutually exclusive")
```

Comment:
Should we also check if one of them is specified as well? What would a KeyValueTablePanel look like if no data is provided?

EDIT: seems like all the other subclasses can be benefited from this change as well.

Reply:
Subclasses might choose to specify neither `data` nor `context_data_key` and instead implement a custom `get_data(context)` function instead. Agreed it's a bit messy to have so many ways to do things though.